server/odc-service/src/main/java/com/oceanbase/odc/service/connection/util/ConnectionInfoUtil.java

```diff
@@ -61,6 +64,12 @@ public static void initConnectionId(@NonNull Statement statement, @NonNull Conne
             String sessionId = queryConnectionId(statement, connectionSession.getDialectType());
             Verify.notNull(sessionId, "SessionId");
             connectionSession.setAttribute(ConnectionSessionConstants.CONNECTION_ID_KEY, sessionId);
+            if (connectionSession.getDialectType().isOceanbase() && VersionUtils.isGreaterThanOrEqualsTo(
+                    ConnectionSessionUtil.getVersion(connectionSession), "4.2")) {
+                String proxySessId =
+                        OBUtils.queryOBProxySessId(statement, connectionSession.getDialectType(), sessionId);
```

Comment:
what if current user has no privilege to get proxySessId ?

Reply:
The proxySessId would be null. 
But if user has no privilege on `oceanbase.*`, the query profile woule be unavailable anyway.